README.txt
SQL-based data consumer for the Tesseract project.

Please refer to the manual for more information

ChangeLog
2013-10-15 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Adjusted SQL definition to fit TYPO3 Instal Tool, resolves #51967

2013-06-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.9.0

2013-06-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added index on cache table for improved performance, resolves #48834

2013-05-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected SQL definition of cache table, resolves #48454
	* Compatibility verified with TYPO3 CMS 6.1, resolves #47517

2013-04-11 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Solved issue with trimming beyond query parts length, resolves #42802

2013-04-01 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Refactored SQL parser test suite, resolves #46846
	* Added test cases for SQL parsing exceptions, resolves #46847
	* Compatibility verified with TYPO3 CMS 6.0, resolves #46851

2012-12-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed abusive overriding of parent method in tx_dataquery_wrapper, resolves #43843
	* Released to TER as version 1.8.2

2012-09-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed error with natural sorting in PHP-side sorting algorithm, resolves #40655
	* Released to TER as version 1.8.1

2012-07-30 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.8.0

2012-07-24 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added handling of "engine" property in ordering, resolves #39193

2012-07-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.7.0

2012-06-29 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added definition of search fields, resolves #38440

2012-06-26 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Replaced all t3lib_div::debug() calls, references #38236

2012-06-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added note in the documentation about the use of brackets in queries, resolves #37151

2012-06-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 4.7, raised requirements to TYPO3 4.5, resolves #38236
	* Cleaned up unused bits of code, completed phpDoc comments
	* Added note in the documentation about static values not being supported in queries, resolves #37179

2012-04-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.6.0

2012-04-03 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed irregular malfunctions of the validation wizard, resolves #35594
	* Added warning about duplicate fields in query, resolves #33061
	* Added support for random ordering from data filter, references #27780

2012-04-02 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added support for special ordering keywords (RAND, NULL), resolves #27780

2012-03-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Updated the process of initializing the empty data structure, resolves #35050

2012-02-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.5.0

2012-02-03 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added possibility to return also id-list type data structures, resolves #33602

2012-01-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Externalized transformation of data filter conditions to SQL, resolves #33317
	* Added support for negated conditions in Data Filters, resolves #33195

2012-01-15 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Improved query parsing and execution logic, resolves #33175
	* Added debug logging with the controller
	* Removed debug configuration option, resolves #33205

2012-01-14 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected wrong cache hash calculation for FE users, resolves #33193

2011-09-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.4.0

2011-09-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added possibility to exclude specific tables from language mechanisms, resolves #30039
	* Corrected broken language overlays on joined tables, resolves #30036
	* Corrected version dependency on datafilter (1.2.0+ required)

2011-08-24 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected filter handling to ignore \all value with all operators, resolves #29235

2011-08-16 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.3.0

2011-08-14 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added a check to warn about missing uid when using DISTINCT in a query, resolves #28969

2011-07-26 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Improved filter unit test to use a dataProvider, resolves #28463
	* Improved unit tests to fully run for multiple tables in query, resolves #13988

2011-07-18 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Used new hook of datafilter to count "additional SQL" into empty filter check, resolves #28287

2011-07-17 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added support for void filters, resolves #28274

2011-05-18 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.2.2

2011-05-16 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added check for possible empty array when checking query fields, resolves #26835
	* Added check to avoid caching in workspaces, resolves #26847

2011-03-23 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected error in usage of table aliases for enable fields condition on sub-tables, resolves #13985
	* Released to TER as version 1.2.1

2011-03-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Improved unit tests to run in various contexts, resolves #12521
	* Corrected detection of ON clause in SQL parser, resolves #13247
	* Added storage of filter information inside data structure, resolves #10753
	* Released to TER as version 1.2.0

2011-01-25 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added workspace/versioning support, resolves #10207
	* Corrected behavior of brackets when rebuilding the query, resolves #11241

2010-12-28 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.1.0

2010-12-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>


ext_emconf.php
<?php

/*********************************************************************
* Extension configuration file for ext "dataquery".
*
* Generated by ext 21-06-2013 12:51 UTC
*
* https://github.com/t3elmar/Ext
*********************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'SQL-based Data Provider - Tesseract project',
  'description' => 'Assembles a query on data stored in the TYPO3 local database, automatically enforcing criteria like language, publication date, etc. More info on http://www.typo3-tesseract.com/',
  'category' => 'misc',
  'author' => 'Francois Suter (Cobweb)',
  'author_email' => 'typo3@cobweb.ch',
  'shy' => '',
  'dependencies' => 'tesseract,datafilter,overlays,expressions',
  'conflicts' => '',
  'priority' => '',
  'module' => '',
  'state' => 'stable',
  'internal' => '',
  'uploadfolder' => 0,
  'createDirs' => '',
  'modify_tables' => '',
  'clearCacheOnLoad' => 0,
  'lockType' => '',
  'author_company' => '',
  'version' => '1.9.0',
  'constraints' => 
  array (
    'depends' => 
    array (
      'tesseract' => '1.5.0-0.0.0',
      'datafilter' => '1.6.0-0.0.0',
      'overlays' => '2.0.0-0.0.0',
      'typo3' => '4.5.0-6.1.99',
      'expressions' => '',
    ),
    'conflicts' => 
    array (
    ),
    'suggests' => 
    array (
      'devlog' => '',
      'cachecleaner' => '',
    ),
  ),
  '_md5_values_when_last_written' => 'a:36:{s:9:"ChangeLog";s:4:"5681";s:27:"class.tx_dataquery_ajax.php";s:4:"2d32";s:28:"class.tx_dataquery_cache.php";s:4:"1106";s:29:"class.tx_dataquery_parser.php";s:4:"a33b";s:34:"class.tx_dataquery_queryobject.php";s:4:"b13b";s:32:"class.tx_dataquery_sqlparser.php";s:4:"f447";s:33:"class.tx_dataquery_sqlutility.php";s:4:"9e86";s:30:"class.tx_dataquery_wrapper.php";s:4:"fc7e";s:16:"ext_autoload.php";s:4:"6a06";s:21:"ext_conf_template.txt";s:4:"bd30";s:12:"ext_icon.gif";s:4:"ebf0";s:17:"ext_localconf.php";s:4:"c03f";s:14:"ext_tables.php";s:4:"28e0";s:14:"ext_tables.sql";s:4:"5115";s:13:"locallang.xml";s:4:"8d70";s:37:"locallang_csh_txdatafilterfilters.xml";s:4:"73bc";s:36:"locallang_csh_txdataqueryqueries.xml";s:4:"d2d2";s:16:"locallang_db.xml";s:4:"fe86";s:10:"README.txt";s:4:"b3a6";s:7:"tca.php";s:4:"cd0f";s:14:"doc/manual.pdf";s:4:"f3da";s:14:"doc/manual.sxw";s:4:"7548";s:14:"doc/manual.txt";s:4:"f4d7";s:43:"hooks/class.tx_dataquery_datafilterhook.php";s:4:"43e3";s:34:"res/icons/add_dataquery_wizard.gif";s:4:"909a";s:39:"res/icons/icon_tx_dataquery_queries.gif";s:4:"ebf0";s:22:"res/js/check_wizard.js";s:4:"b347";s:42:"samples/class.tx_dataquery_sample_hook.php";s:4:"6530";s:34:"tests/tx_dataquery_parser_Test.php";s:4:"bdb8";s:46:"tests/tx_dataquery_sqlbuilder_default_Test.php";s:4:"8bf8";s:47:"tests/tx_dataquery_sqlbuilder_language_Test.php";s:4:"5b9c";s:38:"tests/tx_dataquery_sqlbuilder_Test.php";s:4:"2255";s:48:"tests/tx_dataquery_sqlbuilder_workspace_Test.php";s:4:"7b25";s:37:"tests/tx_dataquery_sqlparser_Test.php";s:4:"1d40";s:35:"tests/tx_dataquery_wrapper_Test.php";s:4:"c609";s:44:"wizards/class.tx_dataquery_wizards_check.php";s:4:"7a93";}',
  'suggests' => 
  array (
  ),
  'comment' => 'Compatibility with TYPO3 6.0 and 6.1; minor bug fixes.',
  'user' => 'francois',
);

?>